crates/biome_js_analyze/src/lint/performance/no_accumulating_spread.rs

```diff
@@ -140,3 +160,47 @@ fn is_known_accumulator(node: &JsSpread, model: &SemanticModel) -> Option<bool>
     // Finally check that the spread references the first parameter.
     Some(parameter.syntax().index() == 0)
 }
+
+fn handle_spread(node: &JsSpread, model: &SemanticModel) -> Option<bool> {
+    let reference = node
+        .argument()
+        .ok()?
+        .as_js_identifier_expression()?
+        .name()
+        .ok()?;
+
+    is_known_accumulator(&reference, model)
+}
+
+// https://github.com/biomejs/biome/issues/5277
+// Spread operators are just syntax for Object.assign
+// Lets handle the rare cases where someone may use this
+fn handle_object_assign(node: &JsStaticMemberExpression, model: &SemanticModel) -> Option<bool> {
+    let object = node.object().ok()?;
+
+    let object_name = object.as_js_identifier_expression()?.name().ok()?;
+    if object_name.to_trimmed_text() != "Object" {
+        return None;
+    }
+    let operator = node.member().ok()?;
+    if operator.to_trimmed_text() != "assign" {
+        return None;
+    }
+
+    let call_expression = node.parent::<JsCallExpression>()?;
+    let arguments = call_expression.arguments().ok()?;
+    let reference = arguments
+        .args()
+        .first()?
+        .ok()?
+        .as_any_js_expression()?
+        .as_js_identifier_expression()?
+        .name()
+        .ok()?;
+
+    is_known_accumulator(&reference, model)
+}
+
+declare_node_union! {
+    pub SpreadOrAssign = JsSpread | JsStaticMemberExpression
+}
```

Comment:
nit: Let move this to above the `impl Rule` for readability

nit: adhere to naming convention
```suggestion
declare_node_union! {
    pub AnySpread = JsSpread | JsStaticMemberExpression
}
```